packages/core/src/main/logger/ipc-file-logger.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { LogEntry, transports } from "winston";
+
+type IpcFileLoggerOptions = Omit<transports.FileTransportOptions, "filename">;
+
+class IpcFileLogger {
+  private fileTransports = new Map<string, transports.FileTransportInstance>();
+
+  constructor(
+    private options: IpcFileLoggerOptions,
+    private createNewFileTransport: (
+      options: transports.FileTransportOptions
+    ) => transports.FileTransportInstance,
+  ) {}
+
+  log({ fileId, entry }: { fileId: string; entry: LogEntry }) {
+    const transport = this.ensureTransportForFile(fileId);
+
+    transport?.log?.(entry, () => {});
+  }
+
+  close(fileId: string) {
+    const transport = this.fileTransports.get(fileId);
+
+    if (transport) {
+      transport.close?.();
```

Comment:
Is the typing of `transport.close` really `undefined | () => void`?

Reply:
Yes unfortunately it is